pkg/nfd-topology-updater/nfd-topology-updater.go

```diff
@@ -94,6 +96,28 @@ func NewTopologyUpdater(args Args, resourcemonitorArgs resourcemonitor.Args, pol
 // one request if OneShot is set to 'true' in the updater args.
 func (w *nfdTopologyUpdater) Run() error {
 	klog.Infof("Node Feature Discovery Topology Updater %s", version.Get())
+	if w.args.GCEnabled {
```

Comment:
I'd dare more. I'd go as far as renaming this function `RunUpdater` and add `RunGarbageCollector` which includes only the code you added here. Then, in the calling site, we can make logic on `w.args.GCEnabled`.

Reply:
Looking at this now I's suggest to make this a totally separate executable, `cmd/nfd-topology-updater-gc` or whatever. The gc functionality is basically a totally separate entrypoint and most of the cmdline flags are ignored for the gc (only kubeconfig is common, afaiu). WDYT?

Reply:
seems like the best approach (and simpler overall)

Reply:
agreed, I will move this to standalone binary.